Add support for FMUv6.

smithandrewc opened the pull request:
USB-HS (ULPI) support still outstanding.

dagar commented:
Can you add `px4_fmu-v6_default` to the builds?
 - https://github.com/PX4/PX4-Autopilot/blob/de6661634161be00c5fd7949875b401e71f21358/.ci/Jenkinsfile-compile#L85
 - https://github.com/PX4/PX4-Autopilot/blob/de6661634161be00c5fd7949875b401e71f21358/.github/workflows/compile_nuttx.yml#L64

smithandrewc commented:
> Can you add `px4_fmu-v6_default` to the builds?
> 
> * https://github.com/PX4/PX4-Autopilot/blob/de6661634161be00c5fd7949875b401e71f21358/.ci/Jenkinsfile-compile#L85
> * https://github.com/PX4/PX4-Autopilot/blob/de6661634161be00c5fd7949875b401e71f21358/.github/workflows/compile_nuttx.yml#L64

Done.

davids5 commented on boards/px4/fmu-v6/src/hw_config.h:
```diff
@@ -0,0 +1,128 @@
+#define INTERFACE_USART                1
+#define INTERFACE_USART_CONFIG         "/dev/ttyS0,115200"
+#define BOOT_DELAY_ADDRESS             0x000001a0
+#define BOARD_TYPE                     53
```
52 is reserved for PX4 FMU V6, but on the SW dev call this board came up and The current thinking as expressed by @dagar is this will need to be moved to a gumstics FMUV6 board and this should be discussed. Fyi @LorenzMeier

smithandrewc replied:
ok thanks.  So is the name changing, from "px4_fmu-v6" to "gumstix_fmu-v6"?  Do you know what the BOARD_TYPE should be set to?

davids5 replied:
I have reserved 54 see https://github.com/PX4/PX4-Bootloader/blob/master/board_types.txt#L42

smithandrewc replied:
@davids5 I've got all the changes made, could you just confirm please that I should be renaming (moving) this from px4/fmu-v6 to gumstix/fmu-v6?   Thanks.

davids5 replied:
@dagar ?

smithandrewc replied:
I made the switch.

davids5 reviewed (commented):
@smithandrewc - this looks Great! One comment on where in the tree and what ID will be assigned.

davids5 commented on boards/px4/fmu-v6/nuttx-config/stackcheck/defconfig:
```diff
@@ -0,0 +1,282 @@
+CONFIG_CDCACM_PRODUCTSTR="PX4 FMU v6.x"
+CONFIG_CDCACM_RXBUFSIZE=600
+CONFIG_CDCACM_TXBUFSIZE=12000
+CONFIG_CDCACM_VENDORID=0x3185
```
@smithandrewc Do you have a gumstix Vendor ID ?

smithandrewc replied:
No I don't believe we have a vendor id.

smithandrewc replied:
Altium has a vendor id, I'm waiting for confirmation if we can use it.

davids5 replied:
I believe ST has a program, but you have to check the details.

smithandrewc replied:
I'm going to use the Altium vendor id 0x1B8C.

davids5 commented on boards/px4/fmu-v6/nuttx-config/stackcheck/defconfig:
```diff
@@ -0,0 +1,282 @@
+CONFIG_BUILTIN=y
+CONFIG_C99_BOOL8=y
+CONFIG_CDCACM=y
+CONFIG_CDCACM_PRODUCTID=0x0035
```
This will be 0x36 (54 dec)

mrpollo commented:
@smithandrewc it looks like you addressed most of the comments already, can you please provide a status update to this PR? 

@davids5 what's missing to get this merged?

smithandrewc commented:
> @smithandrewc it looks like you addressed most of the comments already, can you please provide a status update to this PR?

PR has been rebased on master.  I believe only outstanding issue is to get USB-HS working.

davids5 commented:
@mrpollo @LorenzMeier, @dagar - this is safe to merge, and we can continue as needed.

LorenzMeier commented on .ci/Jenkinsfile-compile:
```diff
@@ -49,6 +49,7 @@ pipeline {
                      "cubepilot_cubeorange_default",
                      "cubepilot_cubeyellow_console",
                      "cubepilot_cubeyellow_default",
+                     "gumstix_fmu-v6u_default",
```
Given this follows the fmu-v6u standard, could you rename this to px4_fmu-v6u_default?

smithandrewc replied:
Absolutely, that's how it was named originally.  I'll rename and push now.

smithandrewc replied:
Updated and pushed.
